### ISSUE TITLE
No package metadata was found for egon.data

### ISSUE THREAD

**Issue by gplssm** [openego/eGon-data#60]
After a fresh install of eGon-data, done with `pip install --use-feature=2020-resolver`, I cannot use the CLI anymore

```
Traceback (most recent call last):
  File "/home/guido/miniconda3/envs/eGon-data-new/bin/egon-data", line 33, in <module>
    sys.exit(load_entry_point('egon.data', 'console_scripts', 'egon-data')())
  File "/home/guido/miniconda3/envs/eGon-data-new/bin/egon-data", line 22, in importlib_load_entry_point
    for entry_point in distribution(dist_name).entry_points
  File "/home/guido/miniconda3/envs/eGon-data-new/lib/python3.7/site-packages/importlib_metadata/__init__.py", line 553, in distribution
    return Distribution.from_name(distribution_name)
  File "/home/guido/miniconda3/envs/eGon-data-new/lib/python3.7/site-packages/importlib_metadata/__init__.py", line 203, in from_name
    raise PackageNotFoundError(name)
importlib_metadata.PackageNotFoundError: No package metadata was found for egon.data
```

**Comment by gnn** [comment]
That's weird. In order to reproduce this on my end, I need the following information:

  - the commit id which you are installing,
  - the output of `pip --version`,
  - the output of `python --version`.

**Comment by gplssm** [comment]
- dev branch 68dc4a1
- `pip --version` pip 20.2.4 from /home/guido/miniconda3/envs/eGon-data-new/lib/python3.7/site-packages/pip (python 3.7)
- `python -V` Python 3.7.9

**Comment by gnn** [comment]
The only update is that I can not reproduce the error.
Using the exact same versions, everything works fine for me.
Did you install `egon.data` with the `-e` flag, or without it? Can I get the output of `pip freeze`?

**Comment by gplssm** [comment]
With `-e`. `pip freeze`

```
alembic==1.4.3
apache-airflow==1.10.12
apispec==1.3.3
appdirs==1.4.4
argcomplete==1.12.2
attrs==19.3.0
Babel==2.9.0
black==20.8b1
cached-property==1.5.2
cattrs==1.0.0
certifi==2020.11.8
chardet==3.0.4
click==7.1.2
click-plugins==1.1.1
cligj==0.7.1
colorama==0.4.4
colorlog==4.0.2
configparser==3.5.3
croniter==0.3.36
defusedxml==0.6.0
dill==0.3.3
dnspython==2.0.0
docutils==0.16
-e git+git@github.com:openego/eGon-data.git@8fe32b85b059f8a2d1656f8515425c07b8b1c277#egg=egon.data
email-validator==1.1.2
Fiona==1.8.18
Flask==1.1.2
Flask-Admin==1.5.4
Flask-AppBuilder==2.3.4
Flask-Babel==1.0.0
Flask-Caching==1.3.3
Flask-JWT-Extended==3.25.0
Flask-Login==0.4.1
Flask-OpenID==1.2.5
Flask-SQLAlchemy==2.4.4
flask-swagger==0.2.14
Flask-WTF==0.14.3
funcsigs==1.0.2
future==0.18.2
GeoAlchemy2==0.6.3
geopandas==0.8.1
graphviz==0.15
gunicorn==20.0.4
idna==2.10
importlib-metadata==3.1.0
iso8601==0.1.13
isort==5.6.4
itsdangerous==1.1.0
Jinja2==2.11.2
json-merge-patch==0.2
jsonschema==3.2.0
lazy-object-proxy==1.5.1
lockfile==0.12.2
Mako==1.1.3
Markdown==2.6.11
MarkupSafe==1.1.1
marshmallow==2.21.0
marshmallow-enum==1.5.1
marshmallow-sqlalchemy==0.23.1
munch==2.5.0
mypy-extensions==0.4.3
natsort==7.1.0
numpy==1.19.4
oedialect==0.0.8
pandas==1.1.4
pathspec==0.8.1
pendulum==1.4.4
prison==0.1.3
psutil==5.7.3
psycopg2==2.8.6
psycopg2-binary==2.8.6
pyaml==20.4.0
Pygments==2.7.2
PyJWT==1.7.1
pyproj==3.0.0.post1
pyrsistent==0.17.3
python-daemon==2.2.4
python-dateutil==2.8.1
python-editor==1.0.4
python-nvd3==0.15.0
python-slugify==4.0.1
python3-openid==3.2.0
pytz==2020.4
pytzdata==2020.1
PyYAML==5.3.1
regex==2020.11.13
requests==2.25.0
setproctitle==1.1.10
Shapely==1.7.1
six==1.15.0
SQLAlchemy==1.3.20
SQLAlchemy-JSONField==0.9.0
SQLAlchemy-Utils==0.36.8
tabulate==0.8.7
tenacity==4.12.0
text-unidecode==1.3
thrift==0.13.0
toml==0.10.2
typed-ast==1.4.1
typing-extensions==3.7.4.3
tzlocal==1.5.1
unicodecsv==0.14.1
urllib3==1.26.2
Werkzeug==0.16.1
WTForms==2.3.3
zipp==3.4.0
zope.deprecation==4.4.0
```

**Comment by gplssm** [comment]
**Update:** omitting flag `-e` the error does not appear and `egon-data serve` runs correctly

**Comment by gplssm** [comment]
**Update 2:** 
Knowing that the issue is only related to editable mode installations (`-e`) and with inspiration from [a SO discussion](https://stackoverflow.com/questions/58842090/running-entry-point-console-script-in-python-development-environment) I deleted the egg files in `src/egon.data.egg-info` (deleted the entire folder). This solves my problem. 

@gnn sorry for wasting your time!

I will close this, once I made sure that this information lands in the installation guide. Best in an faq/trouble shooting section.

**Comment by gnn** [comment]
Since I ran into the same issue. And the only thing that fixed it was updating `importlib-metadata` which is now at version `3.1.1`. Downgrading to `3.1.0` breaks things again.

**Comment by gnn** [comment]
So yeah, python/importlib_metadata#261 references your error message, is comparable to our case, at least in the sense that we also have a namespace package, and its fix got merged into the latest release which is roughly a day old. So I guess this is a rare case of an upstream bug. Note also, that `importlib-metadata` is a backport of functionality from current Python versions to Python 3.7, i.e. the package usually doesn't need to be installed when using Python versions >= 3.8.

**Comment by gplssm** [comment]
> So yeah, [python/importlib_metadata#261](https://github.com/python/importlib_metadata/issues/261) references your error message, is comparable to our case, at least in the sense that we also have a namespace package, and its fix got merged into the latest release which is roughly a day old. So I guess this is a rare case of an upstream bug. Note also, that `importlib-metadata` is a backport of functionality from current Python versions to Python 3.7, i.e. the package usually doesn't need to be installed when using Python versions >= 3.8.

Wow! Didn't expect that. Thanks for researching!

With the extended knowledge, I'd propose to mention the current workaround (see https://github.com/openego/eGon-data/issues/60#issuecomment-737167646) in "Installation guide > Trouble shooting" an. We can delete that once `,importlib-metadata >3.1.0` is available on PyPi and has n conflicts with other packages in the requirements.

For now, we should keep support for Python 3.7, but we can consider to drop this support in the feature in order to spend less time on supporting many versions.

**Comment by gnn** [comment]
> We can delete that once `,importlib-metadata >3.1.0` is available on PyPi and has n conflicts with other packages in the requirements.

That's already the case. It's just really new. Like, from today. This is why it worked initially for me when I tested it on a fresh virtual environment and only started breaking when I forcefully installed the versions from your `pip freeze` output. So I guess we can close the bug. It's unlikely to resurface again and even if, pointing people towards the solution isn't much trouble.

**Comment by gplssm** [comment]
With the requesting to include this aspect in FAQ in #65, I close this issue.
